### ISSUE TITLE
Signed distance query should support threshold for better performance

### ISSUE THREAD

**Issue by calderpg-tri** [RobotLocomotion/drake#10934]
Right now, collision checks using SceneGraph use one of two options:

- `ComputePointPairPenetration` for binary checks

- `ComputeSignedDistancePairwiseClosestPoints` when gradients or padding are required

Using `ComputeSignedDistancePairwiseClosestPoints` is significantly more expensive, since it performs N^2 checks, but is the only way to implement collision avoidance in controllers or variable padding around the robot/environment.

In practice, many uses of `ComputeSignedDistancePairwiseClosestPoints` look like this:
```
const auto& query_object = GetQueryObject();
std::vector<SignedDistancePair<double>> signed_distance_pairs = query_object.ComputeSignedDistancePairwiseClosestPoints();
for (const auto& signed_distance_pair : signed_distance_pairs) {
  const double distance = signed_distance_pair.distance;
    if (distance <= influence_distance) {
      // do work
    }
}
```
where only those pairs within an influence distance are used.

We should offer a version of `ComputeSignedDistancePairwiseClosestPoints()` that takes an influence distance parameter such as `ComputeSignedDistancePairwiseClosestPointsWithinDistance(double distance)` so that we can take advantage of broadphase to reduce the expense of collision checking.

**Comment by calderpg-tri** [comment]
Marked as high priority since this affects RobotBridge collision avoidance and the performance of all planners in anzu.

**Comment by sherm1** [comment]
This makes a lot of sense and seems like it would be easy to add? @DamrongGuoy built that feature 
into the third available query:
```c++
  std::vector<SignedDistanceToPoint<double>>
  ComputeSignedDistanceToPoint(const Vector3<double>& p_WQ, double threshold);
```

Assigned to @DamrongGuoy and @SeanCurtis-TRI for investigation/discussion.

**Comment by tri-ltyyu** [comment]
Important for dish demo. Have benchmark use cases for testing. Should be low amount of work.

**Comment by jamiesnape** [comment]
(As fundamental as collision checking is, this is surely high priority for a toolkit such as Drake regardless of Anzu.)

**Comment by SeanCurtis-TRI** [comment]
@jamiesnape FTR the issue name is *slightly* misleading. This isn't strictly about performance of the collision checking system. It's about a paucity of options in the API. In other word, @calderpg-tri has been forced to swat flies with a sledge hammer. Adding the parameter won't affect users that do collision detection, or users that require O(N^2) signed distance results. So, it's not a general "performance" tweak that helps the world at large. However, it is a valuable API with good utility.

**Comment by jamiesnape** [comment]
> the issue name is slightly misleading

Would be good to fix that. Does not look so great on the first page of issues for someone looking at the toolkit for the first time...

**Comment by sherm1** [comment]
Good idea @jamiesnape, done.

**Comment by SeanCurtis-TRI** [comment]
Resolved by #11429.

@calderpg-tri can you confirm and close this issue?

**Comment by SeanCurtis-TRI** [comment]
Exercising this new functionality seems to have [revealed a problem](https://github.awsinternal.tri.global/robotics/enabling-anzu/pull/2318). Won't be closing this PR until that is resolved.

**Comment by jamiesnape** [comment]
Can you summarize the problem instead of posting an intranet link?

**Comment by SeanCurtis-TRI** [comment]
@jamiesnape sorry about that.

Essentially, for a given configuration, using a threshold of zero, two passes on the what should be the same data produce different answers. I'm not entirely clear on the full characterization of the bug. But that's the crux.
